pkg/webhook/storageclass.go

```diff
@@ -165,11 +109,16 @@ func applyV1StorageClassPatch(sc *storagev1.StorageClass) *v1.AdmissionResponse
 		return reviewResponse
 	}
 
-	if _, ok := sc.Parameters["instancePrefix"]; ok {
+	tier, ok := sc.Parameters["tier"]
+	if !ok || tier != TierEnterprise {
+		return rejectV1AdmissionResponse(fmt.Errorf("mutlishare is only supported on %q tier instances", TierEnterprise))
+	}
+
+	if _, ok := sc.Parameters[InstanceStorageClassLabel]; ok {
 		return reviewResponse
 	}
 
-	scPatch := fmt.Sprintf(`[{"op":"add", "path":"/parameters/instancePrefix","value": "%s"}]`, sc.Name)
+	scPatch := fmt.Sprintf(`[{"op":"add", "path":"/parameters/%s","value": "%s"}]`, InstanceStorageClassLabel, sc.Name)
```

Comment:
we should also check for valid regex of the storage class name i think. The storage class name will eventually be stamped as a label. It should comply with the value reqs of label: https://cloud.google.com/filestore/docs/managing-labels#requirements

Reply:
https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/master/pkg/util/util.go#L103

Reply:
done